app/controllers/HasCorrectGoodsController.scala

```diff
@@ -92,20 +101,92 @@ class HasCorrectGoodsController @Inject() (
   // TODO - this is still not functional, it is just to create the url. Implement this properly
   def onSubmitLongerCommodityCode(mode: Mode, recordId: String): Action[AnyContent] =
     (identify andThen getData andThen requireData).async { implicit request =>
+      val submitAction = routes.HasCorrectGoodsController.onSubmitLongerCommodityCode(mode, recordId)
+
       form
         .bindFromRequest()
         .fold(
           formWithErrors =>
             //TODO - use the UpdateCommdityQuery (with recordId) here when available
-            request.userAnswers.get(CommodityQuery) match {
-              case Some(commodity) => Future.successful(BadRequest(view(formWithErrors, mode, commodity)))
+            request.userAnswers.get(LongerCommodityQuery(recordId)) match {
+              case Some(commodity) => Future.successful(BadRequest(view(formWithErrors, commodity, submitAction)))
               case None            => Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad().url))
             },
           value =>
-            for {
-              updatedAnswers <- Future.fromTry(request.userAnswers.set(HasCorrectGoodsPage, value))
-              _              <- sessionRepository.set(updatedAnswers)
-            } yield Redirect(navigator.nextPage(HasCorrectGoodsPage, mode, updatedAnswers))
+            if (value) {
+              for {
```

Comment:
wonder if it's worth putting the logic in this for comprehension into a function

Reply:
Certainly neatens things up . Done